File: 30daysOFcode_Go/day23.go
package main

import (
    "fmt"
)

type Node struct{
	data int
	left *Node
	right *Node
}

func createNode(data int) *Node{
	node := new(Node)
	node.data = data

	return node
}

func insert(root *Node, data int) *Node{
	if(root == nil){
		return createNode(data)
	} 

	var cur *Node
	if(data <= root.data){
		cur = insert(root.left, data);
		root.left = cur;
	} else{
		cur = insert(root.right, data);
		root.right = cur;
	}

	return root;

}

func levelOrder(root *Node) {
	if(root == nil){
		return
	}
	queue := make([] *Node, 0)
	queue = append(queue, root)
	
	for len(queue) > 0 {
		cur := queue[0]
		queue = queue[1:]
		fmt.Printf("%d ", cur.data)

		if cur.left != nil{
			queue = append(queue, cur.left)
		}
		if cur.right != nil{
			queue = append(queue, cur.right)
		}
	}
	fmt.Println("")
}

func main() {
	var n, data int
	var root *Node

	fmt.Scanf("%d", &n)

	for n > 0{
		fmt.Scanf("%d", &data)
		root = insert(root, data)
		n--
	}
	levelOrder(root)
}

File: 30daysOFcode_C#/day 23.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using System.Linq;

class Node{
    public Node left,right;
    public int data;
    public Node(int data){
        this.data=data;
        left=right=null;
    }
}

class Solution{

	static void levelOrder(Node root){
  		Queue<Node> nodes = new Queue<Node>();
        
        nodes.Enqueue(root);

        while(nodes.Count > 0) {
            Node src = nodes.Dequeue();

            Console.Write(src.data + " ");

            if(src.left != null) nodes.Enqueue(src.left);
            if(src.right != null) nodes.Enqueue(src.right);
        }
    }

	static Node insert(Node root, int data){
        if(root==null){
            return new Node(data);
        }
        else{
            Node cur;
            if(data<=root.data){
                cur=insert(root.left,data);
                root.left=cur;
            }
            else{
                cur=insert(root.right,data);
                root.right=cur;
            }
            return root;
        }
    }
    static void Main(String[] args){
        Node root=null;
        int T=Int32.Parse(Console.ReadLine());
        while(T-->0){
            int data=Int32.Parse(Console.ReadLine());
            root=insert(root,data);            
        }
        levelOrder(root);
        
    }
}
